src/Paramore.Brighter.MessagingGateway.Kafka/KafkaDefaultMessageHeaderBuilder.cs

```diff
@@ -42,38 +43,72 @@ public Headers Build(Message message)
         {
             var headers = new Headers
             {
+                new Header(HeaderNames.MESSAGE_ID, message.Header.MessageId.ToByteArray()),
                 new Header(HeaderNames.MESSAGE_TYPE, message.Header.MessageType.ToString().ToByteArray()),
                 new Header(HeaderNames.TOPIC, message.Header.Topic.Value.ToByteArray()),
-                new Header(HeaderNames.MESSAGE_ID, message.Header.MessageId.ToByteArray()),
             };
 
-            string timeStampAsString = DateTimeOffset.UtcNow.DateTime.ToString(CultureInfo.InvariantCulture);
-            if (message.Header.TimeStamp.DateTime != default)
+            if (message.Header.Bag.TryGetValue(BrighterHeaderNames.UseCloudEvents, out var val) && val is true)
```

Comment:
If there is a string in the header names collection, we should not use a test to switch how we do this. If we want a boolean value, we should use one.  That implies that we have a flag in the message header that indicates whether we use cloud events. This can be set in the mapper, or, via the UseCloudEvents attribute. In that way, the mapper determines how we write/read which is as it should be